File: class23_homework/A-Anton-Georgiev-3/RUBY/Plane.rb
class Plane
	attr_accessor :seats
	
	def initialize
		self.seats = Array.new(54) {Array.new(3)}
	end
	
	def write_seats
		i=0
		while i<54
			write i
			print " "
			write i+1
			print "\n"
		end
	end
	
	def fill_seats(passengers)
		54.times do |i|
			if passengers==3
				if seats[i][0]==0 && seats[i][1]==0 && seats[i][2]==0
					seats[i][0]=1
					seats[i][1]=1
					seats[i][2]=1
					return true
				end
			elsif passengers==2
				if seats[i][0]==0 && seats[i][1]==0
					seats[i][0]=1
					seats[i][1]=1
					return true
				elsif seats[i][1]==0 && seats[i][2]==0
					seats[i][1]=1
					seats[i][2]=1
					return true
				end
			elsif passengers==1
				if seats[i][0]==0
					seats[i][0]=1
					return true
				elsif seats[i][1]==0
					seats[i][1]=1
					return true
				elsif seats[i][2]==0
					seats[i][2]=1
					return true
				end
			else
				return false
			end
		end
	end
	
private
	def write(i)
		3.times do |j|
			print self.seats[i][j]
		end
	end
end

File: class23_homework/A-Anton-Georgiev-3/JAVA/PlaneOrder/src/homework/Plane.java
package homework;

public class Plane {
	private int[][] seats = new int[54][3];

	public int[][] getSeats() {
		return seats;
	}

	public void setSeats(int[][] seats) {
		this.seats = seats;
	}
	
	private void write(int seats[][],int i) {
		for(int j=0;j<3;j++) {
			System.out.print(seats[i][j]);
		}
	}
	
	public void writeSeats() {
		for(int i=0;i<54;i+=2) {
			write(seats,i);
			System.out.print(" ");
			write(seats,i+1);
			System.out.println();
		}
	}
	
	public boolean fillSeats(int passengers) {
		for(int i=0;i<54;i++) {
			if(passengers==3) {
				if(seats[i][0]==0 && seats[i][1]==0 && seats[i][2]==0) {
					seats[i][0]=1;
					seats[i][1]=1;
					seats[i][2]=1;
					return true;
				}
			}
			
			if(passengers==2) {
				if(seats[i][0]==0 && seats[i][1]==0) {
					seats[i][0]=1;
					seats[i][1]=1;
					return true;
				} else if(seats[i][1]==0 && seats[i][2]==0) {
					seats[i][1]=1;
					seats[i][2]=1;
					return true;
				}
			}
			
			if(passengers==1) {
				if(seats[i][0]==0) {
					seats[i][0]=1;
					return true;
				} else if(seats[i][1]==0) {
					seats[i][1]=1;
					return true;
				} else if(seats[i][2]==0) {
					seats[i][2]=1;
					return true;
				}
			}
			return false;
		}
	}
}
